content/blog/2020-03-tibble-3-0-0/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+---
+title: tibble 3.0.0
+slug: tibble-3-0-0
+description: >
+    tibble 3.0.0 is on CRAN now! Tibbles are a modern reimagining of the data frame, keeping what time has shown to be effective, and throwing out what is not, with nicer default output too! This article describes the latest major release and provides an outlook on further developments
+date: 2020-03-30
+author: Kirill Müller
+photo:
+  url: https://unsplash.com/photos/dbOV1qSiL-c
+  author: Vinicius Amano
+categories:
+- package
+- programming
+---
+
+
+```{r setup, include = FALSE}
+library(tidyverse)
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>", error = TRUE)
+```
+
+
+Version 3.0.0 of the tibble package is on CRAN now. Tibbles are a modern reimagining of the data frame, keeping what time has shown to be effective, and throwing out what is not, with nicer default output too! Grab the latest version with:
+
+```r
+install.packages("tibble")
+```
+
+Tibble now fully embraces vctrs, using it under the hood for its subsetting and subset assignment ("subassignment") operations.
+Accessing and updating rows and columns is now based on a rock-solid framework and works consistently for all types of columns, including list, data frame, and 
+matrix columns.
+We believe that the changes will ultimately lead to better and safer code.
+
+This major release required quite some preparation, including a [new vignette](https://www.tidyverse.org/articles/2018/11/tibble-2.0.0-pre-announce/) that described the behavior of subsetting operations and the reasoning behind it.
+Due to the nature of the changes, about 60 CRAN packages were failing with our release candidate.
+Thanks to the maintainers of downstream packages who were very helpful in making this upgrade a smooth experience.
+
+In this blog post, I focus on a few user- and programmer-related changes, and give an outlook over future development:
+
+- How to ensure that an object can be part of a tibble
+- Sturdy recycling
+- Pitfalls with type conversion
+- Safe error handling
+- Formatting
+
+For a complete overview please see the [release notes](https://github.com/tidyverse/tibble/releases/tag/v2.0.0).
+
+Use the [issue tracker](https://github.com/tidyverse/tibble/issues) to submit bugs or suggest ideas, your contributions are always welcome.
+
+
+## What can be part of a tibble?
+
+Tibbles and data frames are collections of columns, each is a vector, all have the same length.
+Neat.
+
+What is a vector?
+What is its length?
+
+The new [vctrs package](https://vctrs.r-lib.org) is dedicated to answering these, surprisingly tricky, questions.
+The `vctrs::vec_is()` function decides if an object is a vector.
+This is important, because some objects are inherently scalar and cannot be added as a column to a data frame.
+
+Environments, functions, and other "special" types of objects are clearly non-vectors.
+It is far less obvious for lists, in particular if they have a `"class"` attribute:
```

Comment:
I think this goes into a bit much depth, and it raises more questions than it answers (i.e. why is POSIXlt consider a vector?)  I'd suggest a framing more like: some things are obviously vectors (integers, characters, ...), and some things are obviously not vectors (environments, functions, ...). Lists are harder because some lists are vectors and some are not. vctrs now uses a heuristic that ...; as well as adding a few special cases from base R.

But I'm not sure that this is even going to be that useful for most users of tibbles.